CameraPanel/CameraAndroid.uno

```diff
@@ -42,24 +42,42 @@ public extern(Android) class Camera
 		cameraAndroid.setOnFrameAvailableListener(new android.graphics.SurfaceTexture.OnFrameAvailableListener() {
 			public void onFrameAvailable(android.graphics.SurfaceTexture surfaceTexture) {
 				if(cameraAndroid.HasSurfaceTexture())
-				{
-					@{Camera:Of(_this).MakeContextCurrent():Call()};
-					surfaceTexture.updateTexImage();
-					@{Camera:Of(_this).OnFrameAvailable():Call()};
-				}
+					@{Camera:Of(_this).DispatchFrameAvailable(Java.Object):Call(surfaceTexture)};
 			}
 		});
 
 		android.app.Activity activity = @(Activity.Package).@(Activity.Name).GetRootActivity();
 		cameraAndroid.start(activity.getWindowManager().getDefaultDisplay().getRotation(), textureHandle, @{Camera:Of(_this).IntFacing:Get()});
 	@}
 
-	void MakeContextCurrent()
+	class FrameAvailableClosure
 	{
-		if defined(Android)
-			extern "GLHelper::MakeWorkerThreadContextCurrent()";
+		Camera _camera;
+		Java.Object _tex;
+
+		public FrameAvailableClosure(Camera camera, Java.Object tex)
+		{
+			_camera = camera;
+			_tex = tex;
+		}
+
+		public void FrameAvailable()
+		{
+			UpdateTexImage(_tex);
```

Comment:
Hi All and thanks a lot for your replies and help,

I tested the following (on my Oppo R9 where the preview works):

1. Launch the app: preview ok
2. App to background > open other apps > reopen ours : **preview resumes ok**
3. App to background > open the mobile's camera app > close it > reopen ours: **preview does not resume**, there is no crash, the latest preview frame keeps being displayed, but not updated.